gptscript/datasets.py

```diff
@@ -0,0 +1,25 @@
+from typing import Dict
+from pydantic import BaseModel
```

Comment:
I decided to use Pydantic for this for easier conversion to and from JSON, and just better feeling types in general.

Reply:
I agree that this is better, but now this is different from everything else here.

Also, how does `pedantic` handle fields that don't exist? For example, if we add a field to any of these in the SDK server, will that render old versions of `py-gptscript` incompatible? I had to add `**kwargs` in various places to fix this particular issue.

Reply:
According to ChatGPT (I'm out of my wheelhouse here lol), it will raise an `AttributeError` if you try to assign to a field that does not exist. Is that a problem in this case? If we end up modifying the fields for datasets, any applications using this SDK that interact with datasets will probably want to be updated so that they still function properly. I'm not sure that backwards compatibility is something we want or need in that case, at least for this part of the SDK.

Reply:
It's not just modifying, but adding brand new fields. For example, adding an `id` field to the `DatasetElement` response will break this.

I agree that anyone using this would want to upgrade to the latest version to get all of the new fields, but should we break them? What are your thoughts?

Reply:
I verified that new fields don't break this.